src/java.base/share/classes/java/lang/foreign/SegmentAllocator.java

```diff
@@ -348,9 +345,14 @@ default MemorySegment allocateArray(ValueLayout.OfChar elementLayout, char... el
      * {@code E*layout.byteSize()} initialized with the provided {@code E} {@code int} {@code elements} as
      * specified by the provided {@code layout} (i.e. byte ordering, alignment and size)}
      *
-     * @implSpec The default implementation of this method first calls {@code this.allocateArray(layout, array.length)}
-     * to allocate a segment after which the provided {@code elements} are written into the segment as
-     * specified by the provided {@code layout}.
+     * @implSpec The default implementation is equivalent to:
+     * {@snippet lang=java :
+     *  int size = Objects.requireNonNull(elements).length;
+     *  MemorySegment seg = allocateArray(Objects.requireNonNull(elementLayout), size);
+     *  MemorySegment.copy(MemorySegment.ofArray(elements), elementLayout, 0,
```

Comment:
there's overloads for copy which accepts the array directly too - so no need to wrap the array into an heap segments - that should simplify the snippet some more